src/tests/system/tests/test_proxy.py

```diff
@@ -67,3 +67,184 @@ def test_proxy__lookup_user_using_pam_ldap_and_nslcd_with_proxy_fast_alias_enabl
+
+    # Configure SSSD with two proxy domains
+    client.sssd.common.proxy("ldap", ["id", "auth"], server_hostname=ldap.host.hostname, domain="domain1.example.com")
+    client.sssd.dom("domain1.example.com")["use_fully_qualified_names"] = "True"
+
+    client.sssd.common.proxy("ldap", ["id", "auth"], server_hostname=ldap.host.hostname, domain="domain2.example.com")
+    client.sssd.dom("domain2.example.com")["use_fully_qualified_names"] = "True"
+
+    # Set domain order with domain1 first
+    client.sssd.sssd.update(domains="domain1.example.com, domain2.example.com")
+
+    # Restart services
+    client.sssd.svc.restart("nslcd")
+    client.sssd.restart()
+
+    # Test FQN resolution
+    result1 = client.tools.getent.passwd(f"{user1.name}@domain1.example.com")
+    assert result1 is not None, "User1 not found with FQN"
+    assert result1.uid == 5000, "User1 has wrong UID"
+    assert result1.home == "/home/user1", "User1 has wrong home directory"
+
+    result2 = client.tools.getent.passwd(f"{user2.name}@domain2.example.com")
+    assert result2 is not None, "User2 not found with FQN"
+    assert result2.uid == 5001, "User2 has wrong UID"
+    assert result2.home == "/home/user2", "User2 has wrong home directory"
+
+    # Test authentication
+    assert client.auth.ssh.password(f"{user1.name}@domain1.example.com", password="Secret123"), "User1 login failed"
+
+    assert client.auth.ssh.password(f"{user2.name}@domain2.example.com", password="Secret123"), "User2 login failed"
+
+    # Verify distinct users by checking each user individually
+    user1_entry = client.tools.getent.passwd(f"{user1.name}@domain1.example.com")
+    user2_entry = client.tools.getent.passwd(f"{user2.name}@domain2.example.com")
+
+    assert user1_entry is not None and user2_entry is not None, "Expected two distinct users"
+    assert user1_entry.uid != user2_entry.uid, "Users have duplicate UIDs"
+
+
+@pytest.mark.importance("low")
+@pytest.mark.topology(KnownTopology.LDAP)
+def test_proxy__case_preserving_handling(client: Client, ldap: LDAP):
+    """
+    :title: Proxy provider preserves original username case with case_sensitive=Preserving
+    :setup:
+        1. Create LDAP user "TestUser" with UID 5003 in ou=users
+        2. Configure SSSD with:
+           - id_provider=proxy
+           - proxy_lib_name=ldap
+           - case_sensitive=Preserving
+        3. Configure nslcd with:
+           - ignorecase=yes (for case-insensitive matching)
+           - validnames regex (to handle special characters)
+        4. Restart SSSD and nslcd services
+    :steps:
+        1. Perform user lookups with different case variants:
+           - getent passwd testuser
+           - getent passwd TESTUSER
+           - getent passwd TestUser
+        2. Verify authentication with different case variants:
+           - Authenticate as testuser
+           - Authenticate as TESTUSER
```

Comment:
Should we also cover the variant that is an exact match (TestUser) too?

Reply:
Added